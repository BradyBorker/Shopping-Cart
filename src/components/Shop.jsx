import { useEffect, useState } from "react";
import Card from "./Card";
import styles from "../styles/shop.module.css"

const Shop = () => {
    const [products, setProducts] = useState(() => JSON.parse(localStorage.getItem('products') ?? '[]'));

    useEffect(() => {
        const getItems = async () => {
            const fetchedProducts = await (await fetch('https://fakestoreapi.com/products?limit=5')).json()

            localStorage.setItem('products', JSON.stringify(fetchedProducts));
            setProducts(fetchedProducts);
        }

        if (products.length === 0) getItems();
    }, [products])

    return (
        <>
            <main className={styles.grid} alt='products'>
                {products.map((product) => (
                    <Card key={product.id} product={product} />
                ))}
            </main>
        </>
    )
}

export default Shop;