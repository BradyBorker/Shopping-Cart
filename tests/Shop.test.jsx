import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import Shop from '../src/components/Shop';

window.fetch = vi.fn(() => {
    const products = [
        { id: 1, title: 'product1', image: '', price: 1, description: '' },
        { id: 2, title: 'product2', image: '', price: 2, description: '' },
    ]

    return Promise.resolve({
        json: () => Promise.resolve(products),
    });
})

describe('Shop component', () => {
    it('renders the Shop page', () => {
        const { container } = render(<BrowserRouter><Shop /></BrowserRouter>);
        expect(container).toMatchSnapshot();
    })

    it('renders the main element', () => {
        render(<BrowserRouter><Shop /></BrowserRouter>);
        expect(screen.getByRole('main')).toBeInTheDocument();
    })

    it('renders two products', () => {
        render(<BrowserRouter><Shop /></BrowserRouter>);
        const main = screen.getByRole('main');
        expect(main.childElementCount).toBe(2);
    })
})