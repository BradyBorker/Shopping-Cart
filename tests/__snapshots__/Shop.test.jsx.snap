// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > renders the Shop page 1`] = `
<div>
  <main
    alt="products"
    class="_grid_ea8c8c"
  />
</div>
`;
